snappi/snappicommon.py

```diff
@@ -217,6 +339,68 @@ def clone(self):
         """Creates a deep copy of the current object
         """
         return self.__deepcopy__(None)
+    
+    def _validate_required(self):
+        """Validates the required properties are set
+        """
+        if getattr(self, "_REQUIRED", None) is None:
+            return
+        for p in self._REQUIRED:
+            if p in self._properties and self._properties[p] is not None:
+                continue
+            msg = "{} is a mandatory property of {}"\
+                " and should not be set to None".format(
+                p, self.__class__
+            )
+            raise ValueError(msg)
+
+    def _validate_types(self):
+        common_data_types = [list, str, int, float, bool]
+        for property, value in self._properties.items():
+            if property not in self._TYPES:
+                continue
+            details = self._TYPES[property]
+            if value is None and property not in self._DEFAULTS and \
+                property not in self._REQUIRED:
+                continue
+            if 'enum' in details and value not in details['enum']:
+                msg = 'property {} shall be one of these' \
+                    ' {} enum, but got {} at {}'
+                raise TypeError(msg.format(
+                    property, details['enum'], value, self.__class__
+                ))
+            if details['type'] in common_data_types and \
+                'format' not in details:
+                if not isinstance(value, details['type']):
```

Comment:
this doesn't handle the scenario where list has items of different types. if not feasible in this PR, please create an issue.

Reply:
we will see the list input only if it is x-pattern, x-pattern does a format which will handle in type validation

Reply:
This isn’t true. One example is name of tx/rx devices in flow.

Reply:
This needs a addition in snappigenerator.py and as well as snappicommon.py will raise issue and address